lib/js/jsfetch.nim

```diff
@@ -0,0 +1,78 @@
+## - Fetch for the JavaScript target: https://developer.mozilla.org/docs/Web/API/Fetch_API
+from asyncjs import PromiseJs
+from httpcore import HttpMethod
+export PromiseJs, HttpMethod
+
+when not defined(js) and not defined(nimdoc):
+  {.fatal: "Module jsfetch is designed to be used with the JavaScript backend.".}
+when defined(nodejs):
+  {.warning: "By design 'fetch()' is defined on the web browser, but may not be on NodeJS.".}
+
+template method2cstring(metod: HttpMethod): cstring =
+  ## Template that takes an `HttpMethod` and returns an *Uppercase* `cstring`,
+  ## but *only* for the HTTP Methods that are supported by the fetch API.
+  ## High performance and minimal code compared to just `$(HttpMethod)`.
+  assert metod notin {HttpTrace, HttpOptions, HttpConnect}, "HTTP Method not supported by fetch API"
+  case metod
+  of HttpHead:   "HEAD".cstring
+  of HttpGet:    "GET".cstring
+  of HttpPost:   "POST".cstring
+  of HttpPut:    "PUT".cstring
+  of HttpDelete: "DELETE".cstring
+  of HttpPatch:  "PATCH".cstring
+  else:          "GET".cstring
+
+func fetch*(url: cstring): PromiseJs {.importcpp: "fetch(#)".}
+  ## `fetch()` API, Simple GET only (generates `fetch(url)`).
+  ##
+  ## .. code-block:: nim
+  ##   doAssert fetch("https://nim-lang.org") is PromiseJs
+
+func fetch*(url: cstring, httpMethod: static[HttpMethod], body: cstring): PromiseJs {.asmNoStackFrame.} =
+  ## `fetch()` API, for any HTTP Method (static overload for optimization).
+  ##
+  ## .. code-block:: nim
+  ##   doAssert fetch("https://my.api.org", HttpPost, "body") is PromiseJs
```

Comment:
I noticed you're doing this a lot: 
proc fun(args): ReturnType
`doAssert fun(args) is ReturnType`
it's a tautology and doesn't add much value. I understand it can't easily run in `nim doc` as a runnableExamples (bc need to run in browser) but can you at least make the `code-block:` example more useful?
eg doing something with the result instead of ` is PromiseJs`

Reply:
I agree.

Reply:
actually, definitely out of scope for this PS, but I think we could make `runnableExamples` work for js in browser mode, using headless mode (at least enable with some options) eg https://developers.google.com/web/updates/2017/06/headless-karma-mocha-chai
would be quite useful to test things where nodejs can't work

Reply:
Nice, feel free to hack on that, sounds interesting. :thinking: